doc: move Thingy:53 getting started guide (#7110)
nrfconnect/sdk-nrf: doc/nrf/ug_thingy53_gs.rst

.. _ug_thingy53_gs:

Getting started with Thingy:53
##############################

This guide helps you get started with the Nordic Thingy:53.
It tells you how to update the firmware on the nRF5340 :term:`System on Chip (SoC)` on the Nordic Thingy:53, and how to get started with machine learning with Edge Impulse.

See :ref:`getting_started` for more information about the |NCS| and the development environment.
For more advanced topics related to the Nordic Thingy:53, such as developing custom firmware, see :ref:`ug_thingy53`.

.. _thingy53_gs_requirements:

Minimum requirements
********************

Make sure you have all the required hardware and that your computer or mobile device has one of the supported operating systems.

Hardware
========

* :term:`Universal Serial Bus (USB)`-C cable
* Computer or mobile device

Software

> ========

Author: Fixed.

If you are using a computer, one of the following operating systems:

* Microsoft Windows 8 or 10

> * macOS X, latest version

Author: Fixed.

* Ubuntu Linux, latest Long Term Support (LTS) version

If you are using a mobile device, one of the following operating systems:

* Android
* iOS

.. _thingy53_gs_installing_software:

Installing the required software
********************************

To work with the Nordic Thingy:53 firmware using a computer, install `nRF Connect for Desktop`_.
After installing and starting the application, install the Programmer app.

To work with the Nordic Thingy:53 firmware using a mobile device, install the `nRF Programmer`_ mobile application from the corresponding application store.

.. _thingy53_gs_updating_firmware:

Updating Nordic Thingy:53 firmware
**********************************

> The Nordic Thingy:53 comes preloaded with Edge Impulse firmware, application firmware, and a SoftDevice controller on the network core of the nRF5340 SoC.

Author: Fixed.

> You can try different samples by updating the firmware on the Nordic Thingy:53 using Bluetooth® Low Energy (LE), USB (MCUboot), or an external debug probe.

Reviewer: ```suggestion
To try different samples, update the firmware on the Nordic Thingy:53 using Bluetooth® Low Energy (LE), USB (MCUboot), or an external debug probe.
```

Author: Very nice. Fixed.

You can update the Nordic Thingy:53 application and network core over Bluetooth LE using the nRF Programmer mobile application for Android or iOS.

Complete these steps to update the firmware:

1. Open the nRF Programmer app.

   A list of available samples appears.

   .. figure:: images/thingy53_sample_list.png
      :alt: nRF Programmer - list of samples

      nRF Programmer - list of samples

#. Select a sample.

   :guilabel:`Application Info` appears.

   .. figure:: images/thingy53_application_info.png
      :alt: nRF Programmer - Application Info

      nRF Programmer - Application Info

#. Select the version of the sample from the drop-down menu.
#. Tap :guilabel:`Download`.

   When the download is complete, the name of the button changes to :guilabel:`Install`.
#. Tap :guilabel:`Install`.

   A list of nearby devices and their signal strengths appears.
#. Select your Nordic Thingy:53 from the list.
   It is listed as :guilabel:`El Thingy:53`.

   The transfer of the firmware image starts, and a progress wheel appears.

   .. figure:: images/thingy53_progress_wheel.png
      :alt: nRF Programmer - progress wheel

      nRF Programmer - progress wheel

>    You can pause the update by tapping :guilabel:`Pause`, or stop the update by tapping :guilabel:`Stop`.

Author: Changed "to" from your suggestion to "if you want to".

   The image transfer is complete when the progress wheel reaches 100%.
   The Nordic Thingy:53 is reset and updated to the new firmware sample.
#. Tap :guilabel:`Done` to return to :guilabel:`Application Info`.

.. _thingy53_gs_updating_usb:

Updating firmware through USB
=============================

> You can update the Nordic Thingy:53 application and network core firmware over USB by using MCUboot, which is a secure bootloader that can be used to update applications without an external debugger.

Reviewer: ```suggestion
You can update the Nordic Thingy:53 application and network core firmware over USB by using MCUboot, which is a secure bootloader that you can use to update applications without an external debugger.
```

Author: Fixed.

.. note::
   Do not unplug the Nordic Thingy:53 during this process.

Complete the following steps to update the firmware:

> 1. Go to `Nordic Thingy:53 Downloads`_.

Author: Fixed.

#. Go to :guilabel:`Edge Impulse firmware`, and download the latest version.
#. Extract the zip file to a location of your choice.
#. Open the connector cover on the side of the Nordic Thingy:53 and plug the Nordic Thingy:53 into the computer using a USB-C cable.

   .. figure:: images/thingy53_sw1_usb.svg
      :alt: The Nordic Thingy:53 schematic - **SW1** and USB connector cover

      The Nordic Thingy:53 schematic - **SW1** and USB connector cover

#. Move the power switch **SW1** to the **ON** position.
#. Open nRF Connect for Desktop and launch the Programmer app.
#. Make sure that :guilabel:`Enable MCUboot` in the :guilabel:`Device` section is enabled.

   .. figure:: images/programmer_enable_mcuboot.png
      :alt: Programmer - Enable MCUboot

      Programmer - Enable MCUboot

#. Move the power switch **SW1** to the **OFF** position.
#. Take off the top cover to access **SW2**.

> #. Press **SW2** while switching **SW1** to the **ON** position.

Author: Huh, **SW2** seems to be a button. Will clarify on the previous line.

   .. figure:: images/thingy53_sw1_sw2.svg
      :alt: The Nordic Thingy:53 schematic - **SW1** and **SW2**

      The Nordic Thingy:53 schematic - **SW1** and **SW2**

#. In the Programmer navigation bar, click :guilabel:`Select device`.

   A drop-down menu appears.
#. In the drop-down menu, select :guilabel:`Bootloader Thingy:53`.
#. Click :guilabel:`Add file` in the :guilabel:`File` section, and select :guilabel:`Browse`.

   A file explorer window appears.

> #. Navigate to where you extracted the Edge Impulse sample firmware.

Author: Fixed.

      Programmer - MCUboot DFU window

#. Click :guilabel:`Write` in the :guilabel:`MCUboot DFU` window.

   The flash slot is erased.
   When the flash slot has been erased, image transfer starts and a progress bar appears.

   When the image transfer has been completed, the network core part of the image is transferred from RAM to the network core flash.
   This can take up to 20 seconds.

   When the update is complete, a :guilabel:`Completed successfully` message appears.

You can now disconnect the Nordic Thingy:53 from the computer.

.. _thingy53_gs_updating_external_probe:

Updating firmware through external debug probe
==============================================

You can update the Nordic Thingy:53 application and network core firmware by using an external debug probe.

.. note::
   The external debug probe must support Arm Cortex-M33, such as the nRF5340 DK.
   You need a 10-pin 2x5 socket-socket 1.27 mm IDC (:term:`Serial Wire Debug (SWD)`) JTAG cable to connect to the external debug probe.

Complete these steps to update the firmware.
In these steps, the nRF5340 DK is used as the external debug probe.
Do no unplug or power off the devices during this process.

> 1. Go to Nordic Thingy:53 Downloads.

Author: Fixed.

#. Go to :guilabel:`Edge Impulse firmware`, and download the latest version.
#. Extract the zip file to a location of your choice.
#. Open nRF Connect for Desktop and launch the Programmer app.
#. Prepare the hardware:
   #. Open the connector cover on the side of the Nordic Thingy:53.
   #. Use a JTAG cable to connect the Nordic Thingy:53 to the debug out port on a 10-pin external debug probe.

      .. figure:: images/thingy53_nrf5340_dk.svg
         :alt: Nordic Thingy:53 connected to the debug port on a 10-pin external debug probe

         Nordic Thingy:53 connected to the debug port on a 10-pin external debug probe

>    #. Power on the Nordic Thingy:53 by moving the power switch **SW1** to the **ON** position.

Author: Fixed.

   #. Power on the external debug probe.
   #. Connect the external debug probe to the computer with a micro-USB cable.

      In the Programmer app's navigation bar, :guilabel:`No devices available` changes to :guilabel:`Select device`.

      .. figure:: /images/programmer_select_device1.png
         :alt: Programmer - Select device

         Programmer - Select device

#. Click :guilabel:`Select device` and select the appropriate debug probe entry from the drop-down list.

   The icon text changes to board name and the SEGGER ID of the selected device, and the :guilabel:`Device memory layout` section indicates that the device is connected.

>    You can identify the nRF5340 DK by its PCA number PCA10095 and its SEGGER ID, which is printed on the label sticker on the DK.

Author: Fixed.

>    You can identify the nRF5340 DK by its PCA number PCA10095 and its SEGGER ID which is printed on the label sticker on the DK.

Author: Fixed.

   If the nRF5340 DK does not show up in the drop-down list, press ``Ctrl+R`` in Windows or ``command+R`` in macOS to restart the Programmer application.

#. Click :guilabel:`Add file` in the :guilabel:`File` section, and select :guilabel:`Browse`.

   A file explorer window appears.

> #. Navigate to where you extracted the Edge Impulse sample firmware.

Author: Fixed.

#. Open the folder that contains the HEX file for the sample that you want to transfer to the Nordic Thingy:53.
#. Select the HEX file and click :guilabel:`Open`.

   The HEX file appears in the :guilabel:`File memory layout` section.
#. Click :guilabel:`Erase & write` in the :guilabel:`Device` section of the side panel.

The update is complete when the animation in the Programmer app's :guilabel:`Device memory layout` section ends.

.. _thingy53_gs_machine_learning:

Getting started with machine learning
*************************************

> The Nordic Thingy:53 is preprogrmammed with Edge Impulse firmware.

Author: Fixed.

> You can connect the Nordic Thingy:53 to the Edge Impulse Studio over Bluetooth LE using the nRF Edge Impulse mobile application or over USB by connecting the Nordic Thingy:53 to a computer.

Author: Restructured whole sentence a little.

The Edge Impulse firmware enables data collection from all the sensors on the Nordic Thingy:53.
The collected data can be used to train and test machine learning models.

> The trained machine learning model can be deployed to the Nordic Thingy:53 through Bluetooth LE or USB.

Author: Fixed.

Complete the following steps to get started with Edge Impulse:

1. Go to the `Edge Impulse`_ website.
#. Create a free Edge Impulse account.
#. Follow the instructions in the `Nordic Semi nRF5340 DK page`_.

.. _thingy53_gs_precompiled_firmware:

Getting started with precompiled firmware samples
*************************************************

> There are multiple samples in the |NCS| that have been precompiled and can be programmed to the Nordic Thingy:53.

Author: Fixed.

Hardware
========

* :term:`Universal Serial Bus (USB)`-C cable
* Computer or mobile device

Software
========

If you are using a computer, one of the following operating systems:

* Microsoft Windows 8 or 10
* macOS, latest version
* Ubuntu Linux, latest Long Term Support (LTS) version

If you are using a mobile device, one of the following operating systems:

* Android
* iOS

.. _thingy53_gs_installing_software:

Installing the required software
********************************

When using a computer to work with with the Nordic Thingy:53 firmware, install `nRF Connect for Desktop`_.
After installing and starting the application, install the Programmer app.

When using a mobile device to work with the Nordic Thingy:53, install the `nRF Programmer`_ mobile application from the corresponding application store.

.. _thingy53_gs_updating_firmware:
.. _thingy53_gs_precompiled_firmware:

Getting started with precompiled firmware samples
*************************************************

The Nordic Thingy:53 is preloaded with Edge Impulse firmware, application firmware, and a SoftDevice controller on the network core of the nRF5340 SoC.

> For more information on the firmware and associated features, see :ref:`ug_thingy53`.

Reviewer 2: Shouldn't we link to Thingy:53's EI page here?

Author: They don't have one, at least at the moment. That's why the instructions in the Getting started with machine learning section refer to the instructions for nRF5340 DK.

* Computer or mobile device

Software
========

If you are using a computer, one of the following operating systems:

* Microsoft Windows 8 or 10
* macOS, latest version
* Ubuntu Linux, latest Long Term Support (LTS) version

If you are using a mobile device, one of the following operating systems:

* Android
* iOS

.. _thingy53_gs_installing_software:

Installing the required software
********************************

To work with the Nordic Thingy:53 firmware:

* using a computer, install `nRF Connect for Desktop`_.
  After installing and starting the application, install the Programmer app.
* using a mobile device, install the `nRF Programmer`_ mobile application from the corresponding application store.

When using a computer to work with with the Nordic Thingy:53 firmware, install `nRF Connect for Desktop`_.
After installing and starting the application, install the Programmer app.

When using a mobile device to work with the Nordic Thingy:53, install the `nRF Programmer`_ mobile application from the corresponding application store.

.. _thingy53_gs_precompiled_firmware:

Getting started with precompiled firmware samples
*************************************************

> The |NCS| contains multiple precompiled samples that you can be program to the Nordic Thingy:53.

Author: Fixed.

   The :file:`CONTENTS.txt` file in the extracted folder contains the location and names of the different firmware images.

#. Open nRF Connect for Desktop and launch the Programmer app.
#. Prepare the hardware:

   #. Open the connector cover on the side of the Nordic Thingy:53.
   #. Use a JTAG cable to connect the Nordic Thingy:53 to the debug out port on a 10-pin external debug probe.

      .. figure:: images/thingy53_nrf5340_dk.svg
         :alt: Nordic Thingy:53 connected to the debug port on a 10-pin external debug probe

         Nordic Thingy:53 connected to the debug port on a 10-pin external debug probe

   #. Power on the Nordic Thingy:53; move the power switch **SW1** to the **ON** position.
   #. Power on the external debug probe.
   #. Connect the external debug probe to the computer with a micro-USB cable.

      In the Programmer app's navigation bar, :guilabel:`No devices available` changes to :guilabel:`Select device`.

      .. figure:: /images/programmer_select_device1.png
         :alt: Programmer - Select device

         Programmer - Select device

#. Click :guilabel:`Select device` and select the appropriate debug probe entry from the drop-down list.

>    The icon text changes to board name and the SEGGER ID of the selected device, and the :guilabel:`Device memory layout` section indicates that the device is connected.

Author: I'll take out the word "SEGGER", but leave ID at least for now.

   You can identify the nRF5340 DK by its PCA number PCA10095 and its SEGGER ID that is printed on the label sticker on the DK.

   If the nRF5340 DK does not show up in the drop-down list, press ``Ctrl+R`` in Windows or ``command+R`` in macOS to restart the Programmer application.

#. Click :guilabel:`Add file` in the :guilabel:`File` section, and select :guilabel:`Browse`.

   A file explorer window appears.
#. Navigate to the folder where you extracted application firmware.
#. Open the :file:`img_app_bl` folder that contains the HEX files used with the debug probe.
#. Select the HEX file for the application that you want to transfer to the Nordic Thingy:53 and click :guilabel:`Open`.

   The HEX file appears in the :guilabel:`File memory layout` section.
#. Click :guilabel:`Erase & write` in the :guilabel:`Device` section of the side panel.

The update is complete when the animation in the Programmer app's :guilabel:`Device memory layout` section ends.

.. _thingy53_gs_updating_firmware:

> Updating Nordic Thingy:53 firmware

Author: Did my best to incorporate the little information that was still in this section into the Getting started with precompiled firmware samples.

Installing the required software
********************************

To work with the Nordic Thingy:53 firmware:

* using a computer, install `nRF Connect for Desktop`_.
  After installing and starting the application, install the Programmer app.
* using a mobile device, install the `nRF Programmer`_ mobile application from the corresponding application store.

When using a computer to work with with the Nordic Thingy:53 firmware, install `nRF Connect for Desktop`_.
After installing and starting the application, install the Programmer app.

> When using a mobile device to work with the Nordic Thingy:53, install the `nRF Programmer`_ mobile application from the corresponding application store.

Author: This section now has two alternatives for presenting the information.

Opinions on which looks better needed.

Reviewer 3: @wiba-nordic I guess this conversation can be resolved?

Author: Yes, I'll take out the bullet point version.
